deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -22,9 +22,6 @@ GrB_Info GxB_MatrixTupleIter_new
 	GB_WHERE(A, "GxB_MatrixTupleIter_new (A)") ;
 	GB_RETURN_IF_NULL_OR_FAULTY(A) ;
 
-	// make sure matrix is not bitmap or full
-	GxB_set(A, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;
-
```

Comment:
Our tuple iterator can only work with SPARSE matrices, in case the matrix format isn't sparse, e.g. bitmap or full the iterator would break,
if I'm not mistaken we use our tuple iterator to iterate over two type of matrices:
1. Graph built-in matrices e.g. label/relationship matrices
2. matrices created as part of a traversal e.g. M = f*A*B

For the first case we can assure the matrices are sparse, for the second, we'll have to call `GxB_set(M, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;` prior to iterating.

I feel like the issue we've encountered is iterating over a built-in matrix, multiple threads creating an iterator over the same matrix all at once.

Reply:
In which case we should change this `GxB_set` to assert on the input matrix format,
in addition to `op_conditional_traverse` there are other areas in the code which are using the iterator e.g. `var_len_traversal`
this entire requirement is soon to be lifted by @OfirMos work on supporting the hyper-sparse format,